### DISCUSSION TITLE
Storage Issues

### DISCUSSION THREAD

**Discussion by usmanaa10** [Security-Onion-Solutions/securityonion#7811] [Unsupported Versions] [answered]
Hi Friends,
            I am working from couple of months but cant solve it, Kindly guide me if you can.
Kindly look at the global configuration I shared in the attachment, the problem is that the  /nsm folder consume all  the storage in 12 days and on 13th day when it was 100% I  reboot the system and the storage  gets clear and /nsm folder consumed space drop to 30%.

I need that the storage starting clearing automatically without rebooting or removing folders in /nsm directory manually.

What kind of changes need to be done in global sls as shared in attachment
Please 
![Secuity Onion Configuration](https://user-images.githubusercontent.com/66419518/163754547-75bd86ba-fa83-40ff-9cf8-277ab09dccad.png)

**Comment by dougburks** [comment]
What is the output of the following?
```
df -h
```
```
sudo du --max-depth=1 -h /nsm
```
```
sudo grep log_size_limit /opt/so/saltstack/local/pillar/minions/*
```

**Reply by usmanaa10** [reply]
Hi @dougburks ,
                  I have attached the output of above given commands in sequence, Kindly have a look to it and guide me where change is required and why.

![1](https://user-images.githubusercontent.com/66419518/164148026-267e4fa6-f020-4a20-85cf-2ccb4cdb0ce3.png)
![2](https://user-images.githubusercontent.com/66419518/164148034-9a081127-047e-46ff-bc8f-351ead5d9746.png)
![3](https://user-images.githubusercontent.com/66419518/164148036-a6532fd1-b7f5-4cd6-8276-cd57472cca43.png)

**Reply by dougburks** [reply]
`/nsm/elasticsearch` should start purging once it reaches 1384GB:
https://docs.securityonion.net/en/2.3/elasticsearch.html#deleting-indices

`/nsm/pcap` should start purging once `/nsm` reaches 90% usage:
https://docs.securityonion.net/en/2.3/stenographer.html#output

You can check log files for additional clues:
`/opt/so/log/sensor_clean.log`
`/opt/so/log/curator/`
`/opt/so/log/stenographer/`

Also, is this a physical machine or virtual machine?

**Comment by usmanaa10** [comment, marked as answer]
Hi @dougburks,
                     Thanks for quick response, 
I have modified  the steno config ,to purge at 75%  but when i restart the service it config automatically change to 90% Why? Not succussed

I have change the log_size_limit = 800     successfully done
![image](https://user-images.githubusercontent.com/66419518/164619672-0ffb3aa9-4883-4745-bdda-f5e7e8ede79e.png)


**Note : My machine is running on VMwareExsi**
![image](https://user-images.githubusercontent.com/66419518/164618112-70ce98af-25de-454f-9e4f-af8de00844a0.png)

 I restart service and then old values are back

![image](https://user-images.githubusercontent.com/66419518/164619309-c1c7e97d-b0ac-4f4d-be8a-c0310f66219e.png)



I dont know whats going on.

My storage reach to above 80% above
![image](https://user-images.githubusercontent.com/66419518/164619561-a3431afd-661f-4d5c-8a78-bec6d4793fae.png)

**Reply by dougburks** [reply]
> I have modified the steno config ,to purge at 75% but when i restart the service it config automatically change to 90% Why? Not succussed

Did you modify the steno config according to the documentation?
https://docs.securityonion.net/en/2.3/stenographer.html#configuration

**Reply by usmanaa10** [reply]
Hi @dougburks ,
                      Thanks for replying.
I made  change in following file
>> cat /opt/so/saltstack/local/pillar/global.sls
![image](https://user-images.githubusercontent.com/66419518/165231611-27e456b7-6a63-4754-b3dd-18a9b4aad77e.png)

>> salt-call state.apply pcap

but pcap files didn't purge 
even I restart the whole machine but 
![image](https://user-images.githubusercontent.com/66419518/165232324-c1647b1f-3603-4c42-ae8f-374f584bd3e0.png)
![image](https://user-images.githubusercontent.com/66419518/165232587-7dc1f032-2f13-4d4a-aa5d-74379eef0f42.png)

**Reply by usmanaa10** [reply]
@dougburks  if you think I made change in a wrong file then let me know where should I made change?

**Reply by usmanaa10** [reply]
Hi @dougburks 
                     After couple of minutes its start working Thanks
Chapter Closed

Thank you soo much
